src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Write.Helpers.cs

```diff
@@ -94,28 +94,37 @@ private static void WriteCore(PooledBufferWriter<byte> output, object value, Typ
         {
             Debug.Assert(type != null || value == null);
 
-            using var writer = new Utf8JsonWriter(output, options.GetWriterOptions());
+            var cachedWriter = CachedUtf8JsonWriter.Get(output, options.GetWriterOptions());
```

Comment:
Don't use `var` here. Same elsewhere, where you can't tell from the right hand side, what the type is, exactly.